### PR TITLE
Parse data from TRSS url

### PR DESCRIPTION
**Pull request by annaibm**
- TKG Parse data from TRSS url

related:https://github.com/adoptium/TKG/issues/558

### CODE REVIEW

**Comment by annaibm** [comment]
Grinder links:
TARGET: sanity.functional
mac:
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/42270/
win:
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/42271/
aarch64_linux:
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/42272/console
s390x_linux:
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/42268/

```
Attempting to get test duration data from TRSS.
12:31:40  curl --silent --max-time 120 -L -k https://trssrtp1.fyre.ibm.com/api/getTestAvgDuration?limit=10&jdkVersion=11&impl=j9&platform=s390x_linux&group=functional&level=sanity -o /home/jenkins/workspace/Grinder/aqa-tests/TKG/response.json
12:31:45  Response data written to file: /home/jenkins/workspace/Grinder/aqa-tests/TKG/response.json
12:31:45  Response Data: {"testListTime":[15134729.416666664],"testLists":[[{"_id":"cmdLineTester_criu_nonPortableRestore_10","avgDuration":2045327.3333333333},{"_id":"testJITServer_2","avgDuration":812512.6666666666},{"_id":"testJITServer_1","avgDuration":704608.6666666666},{"_id":"testJITServer_0","avgDuration":703974.3333333334},{"_id":"cmdLineTester_criu_nonPortableRestore_3","avgDuration":297062},{"_id":"cmdLineTester_criu_nonPortableRestore_4","avgDuration":281803.3333333333},{"_id":"cmdLineTester_GCRegressionTests_3","avgDuration":277104.3333333333},{"_id":"cmdLineTester_GCRegressionTests_2","avgDuration":266454.3333333333},{"_id":"cmdLineTester_GCRegressionTests_1","avgDuration":264064.3333333333},{"_id":"cmdLineTester_GCRegressionTests_0","avgDuration":263644},{"_id":"cmdLineTester_criu_nonPortableRestore_9","avgDuration":235436},{"_id":"cmdLineTester_criu_nonPortableRestore_7","avgDuration":220314.33333333334},{"_id":"cmdLineTester_criu_nonPortableRestore_0","avgDuration":207269.33333333334},{"_id":"cmdLineTester_criu_nonPortableRestore_2","avgDuration":198845},{"_id":"cmdLineTester_criu_nonPortableRestore_5","avgDuration":198723.66666666666},
```

File: src/org/testKitGen/TestDivider.java [hunk 1]
```diff
@@ -226,21 +227,58 @@ private Map<String, Integer> getDataFromTRSS()  {
...
-		String command = "curl --silent --max-time 120 -L " + URL;
+		String osName = System.getProperty("os.name").toLowerCase();
+		String responseFilePath = new File("response.json").getAbsolutePath();
+		String command;
+
+		if (osName.contains("win")) {
+			command = "cmd.exe /c curl --silent --max-time 120 -L -k \"" + URL + "\" -o " + responseFilePath;
+		} else {
+			command = "curl --silent --max-time 120 -L -k "+ URL + " -o " + responseFilePath;
+		}
```
Comment (llxia) [inline src/org/testKitGen/TestDivider.java:238]: A file is suggested to debug the problem. We should try to avoid using it if possible.

**Review by llxia — CHANGES_REQUESTED** [review]

File: src/org/testKitGen/TestDivider.java (hunk 1 above)
Comment (annaibm) [reply src/org/testKitGen/TestDivider.java:238]: I have removed the use of the file `response.json` in the code as suggested. 
Grinder links:
TARGET: sanity.functional
mac:
[42294](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/42294/)
win:
[42295](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/42295/)
aarch64_linux:
[#42296](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/42296/)
s390x_linux:
[#42292](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/42292/)

Comment (llxia) [reply src/org/testKitGen/TestDivider.java:238]: Thanks @annaibm . It looks like the fix is to add quotes around `URL`. Do we need the following changes (L244 - L319)?

Comment (annaibm) [reply src/org/testKitGen/TestDivider.java:236]: I have removed changes from (L244 - L319) . The fix for constructing the curl command based on the operating system . Added quotes around the URL for Windows, while no quotes are used for Linux. Also added the -k option in the curl command.

Comment (annaibm) [reply src/org/testKitGen/TestDivider.java:236]: Grinder links:
x86_64_linux
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/42341/console

s390x_linux
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/42340/

ppc64_aix
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/42338/console

aarch64_linux
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/42344/console

x86_64_windows
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/42343/consoleFull

x86_64_mac
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/42342/console

File: src/org/testKitGen/TestDivider.java [hunk 2]
```diff
@@ -238,7 +246,7 @@ private Map<String, Integer> getDataFromTRSS()  {
 		}
 		try	(InputStream responseStream = process.getInputStream();
 			Reader responseReader = new BufferedReader(new InputStreamReader(responseStream))) {
-			parseDuration(responseReader, map);
+				parseDuration(responseReader, map);
```
Comment (llxia) [inline src/org/testKitGen/TestDivider.java:249]: Could you remove the extra tab?

Comment (annaibm) [reply src/org/testKitGen/TestDivider.java]: Removed the tab.

**Review by llxia — APPROVED** [review]
LGTM

**Review by LongyuZhang — APPROVED** [review]
LGTM. Thanks @annaibm !

**Pull request merged**